core/document/src/intersection.rs

```diff
@@ -1,34 +1,60 @@
-use glam::DVec2;
+use std::ops::Mul;
+
+use glam::{DAffine2, DVec2};
 use kurbo::{BezPath, Line, PathSeg, Point, Shape, Vec2};
 
+#[derive(Debug, Clone, Default, Copy)]
+pub struct Quad([DVec2; 4]);
+
+impl Quad {
+	pub fn from_box(bbox: [DVec2; 2]) -> Self {
+		let size = bbox[1] - bbox[0];
+		Self([bbox[0], bbox[0] + size * DVec2::X, bbox[0] + size * DVec2::Y, bbox[1]])
+	}
+	pub fn lines(&self) -> [Line; 4] {
+		[
+			Line::new(to_point(self.0[0]), to_point(self.0[1])),
+			Line::new(to_point(self.0[1]), to_point(self.0[2])),
+			Line::new(to_point(self.0[2]), to_point(self.0[3])),
+			Line::new(to_point(self.0[3]), to_point(self.0[0])),
+		]
+	}
+}
+
+impl Mul<Quad> for DAffine2 {
+	type Output = Quad;
+
+	fn mul(self, rhs: Quad) -> Self::Output {
+		let mut output = Quad::default();
+		for (i, point) in rhs.0.iter().enumerate() {
+			output.0[i] = self.transform_point2(*point);
+		}
+		output
+	}
+}
+
 fn to_point(vec: DVec2) -> Point {
 	Point::new(vec.x, vec.y)
 }
 
-pub fn intersect_quad_bez_path(quad: [DVec2; 4], shape: &BezPath, closed: bool) -> bool {
-	let lines = vec![
-		Line::new(to_point(quad[0]), to_point(quad[1])),
-		Line::new(to_point(quad[1]), to_point(quad[2])),
-		Line::new(to_point(quad[2]), to_point(quad[3])),
-		Line::new(to_point(quad[3]), to_point(quad[0])),
-	];
+pub fn intersect_quad_bez_path(quad: Quad, shape: &BezPath, closed: bool) -> bool {
 	// check if outlines intersect
 	for path_segment in shape.segments() {
-		for line in &lines {
-			if !path_segment.intersect_line(*line).is_empty() {
+		for line in quad.lines() {
+			if !path_segment.intersect_line(line).is_empty() {
```

Comment:
I think this `for` and `if` could be refactored to just an `if` using `Iterator::any`